File: Typescript/Async_example.ts
const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

const fetchData = async (taskName, delaySeconds) => {
  console.log(`${taskName}: started (will take ${delaySeconds}s)`);
  await sleep(delaySeconds * 1000);
  console.log(`${taskName}: finished`);
  return `${taskName} result`;
};

const main = async () => {
  const start = performance.now();

  const results = await Promise.all([
    fetchData('Task A', 2),
    fetchData('Task B', 1),
    fetchData('Task C', 3),
  ]);

  const end = performance.now();

  console.log('\nResults:');
  for (const result of results) {
    console.log(result);
  }

  console.log(`\nTotal time: ${((end - start) / 1000).toFixed(2)} seconds`);
};

main().catch((error) => {
  console.error('Unexpected error:', error);
});

File: Python/Async_example.py
import asyncio
import time


async def fetch_data(task_name: str, delay: int) -> str:
    print(f"{task_name}: started (will take {delay}s)")
    await asyncio.sleep(delay)
    print(f"{task_name}: finished")
    return f"{task_name} result"


async def main() -> None:
    start = time.perf_counter()

    results = await asyncio.gather(
        fetch_data("Task A", 2),
        fetch_data("Task B", 1),
        fetch_data("Task C", 3),
    )

    end = time.perf_counter()

    print("\nResults:")
    for result in results:
        print(result)

    print(f"\nTotal time: {end - start:.2f} seconds")


if __name__ == "__main__":
    asyncio.run(main())
